pkg/localprovider/config.go

```diff
@@ -0,0 +1,20 @@
+package localprovider
+
+import "fmt"
+
+type LocalProviderConfig struct {
+	ConfigManager ConfigManager `yaml:"configManager"`
+}
+
+type ConfigManager struct {
+	LocalConfigName string `yaml:"localConfigName"`
+}
+
+func (lp *LocalProviderConfig) IsValid() error {
+	cfg := lp.ConfigManager.LocalConfigName
+	if cfg != UPGRADECONFIG_CR_NAME {
+		return fmt.Errorf("Please use %s as the upgrade config name.", UPGRADECONFIG_CR_NAME)
```

Comment:
I definitely get why this is included - the localprovider needs to know the CR name in order to actually work, and it needs to match what the UC Manager thinks the name is. In its current form as a config item it has a few design smells, though - since the value can never deviate from what the default is, it's not a very 'configurable' item. I think the design here can be tweaked slightly to overcome that. For example, what if:

- We still retain a 'CR Name' configmap item, but name it something that's not "local" specific, ie. `upgradeConfigName` or something. 
- The *UpgradeConfigManager* should read in this 'CR Name' configmap value too, when it is initialized. If it's present, it overrides the default `UPGRADECONFIG_CR_NAME` as the name of the CR used by the UpgradeConfigManager when doing stuff. (The UC controller would also have to be updated as part of this change.)
- After doing the above - there is no reason that we need to have the check in this file to make sure that the value in config matches the default, and we can make the CR name a fully configurable item.

As it will be good to get this PR done and merged before your absence, I'm happy to leave this as-is and work on that in OSD-6918 if you like. Do let me know what you think though.

Reply:
I agree it should be better to make this value configurable. But I am not sure if it is in the same scope with OSD-6918? Maybe should better use another card to track?

Reply:
Yeah, good point. It's probably better suited to another card.

Reply:
https://issues.redhat.com/browse/OSD-7073 new card created to track this